### DISCUSSION TITLE
libpython not found on Arch Linux with Python 3.9?

### DISCUSSION THREAD

**Discussion by juliavdkris** [Vector35/binaryninja-api#2184] [General] [answered]
Hi! I purchased Binary Ninja a while ago, and it's been working great so far! But I booted it up today and was greeted with some errors about not being able to find libpython, which I think is related to Arch finally upgrading to Python 3.9 about a week ago. I can see there is a `/usr/lib/libpython3.9.so` file (which is symlinked to `/usr/lib/libpython3.9.so.1.0`), but `pkgconf --libs python3` doesn't list anything. I haven't been able to find any people with similar issues. Does anyone here have an idea on how to troubleshoot this? :)

I am on the latest development build by the way, which according to the change log should have support for Python 3.9

![](https://user-images.githubusercontent.com/22723225/101483764-f083b200-3958-11eb-8f72-f61f74675b39.png)

EDIT: Starting Binary Ninja with the LD_PRELOAD environment variable (`LD_PRELOAD=/usr/lib/libpython3.9.so.1.0 ./binaryninja`) does seem to work. So I guess I can just add that to my KDE application menu shortcut. It would still be nice to know how to properly fix this though, as it might be a problem in the future.

**Comment by psifertex** [comment, marked as answer]
You need to manually specify the interpreter in settings (search for "interpreter") and point it at the `.so`.

We try to update the list of default locations and names to detect so we can probably add that as well, but this work-around is often required for platforms that we don't officially support such as arch since we don't test them regularly.

**Comment by juliavdkris** [comment]
Yep that worked, thank you!

It did suggest the correct path by the way, so I'm guessing it was indeed in one of the standard locations. Could it be that this setting just isn't automatically updated when the system's Python version is upgraded? Perhaps that is intended behavior, but I thought I'd just let you know :)

**Reply by psifertex** [reply]
Ahh, yes, if you've manually selected a pervious python your settings file is updated but then if a system update changes it, you'll need to update it. If you've never manually set one, it should automatically find an appropriate one. You can see if the setting has been changed based on whether `[Modified in User]` shows up. You can also search for all modified settings with `@user` or look in your [settings file](https://docs.binary.ninja/getting-started.html#user-folder:~:text=settings.json%3A%20User%20settings%20file%20(see%20settings))

**Reply by juliavdkris** [reply]
That's strange, because I'm pretty sure I hadn't manually set a Python version before (otherwise I would've known about the setting and just changed it after updating). Oh well, it's working now ¯\\\_(ツ)_/¯
